### ISSUE TITLE
new io: amip_interp bug

### ISSUE THREAD

**Issue by uramirez8707** [NOAA-GFDL/FMS#212]
**Describe the bug**
I run into the following crash in my land only test case with the new io:

FATAL from PE 46: lookup_es: saturation vapor pressure table overflow, nbad= 63

fms_lad2_xanadu.x  000000000262F5E2  mpp_mod_mp_mpp_er          69  mpp_util_mpi.inc
fms_lad2_xanadu.x  0000000002636240  mpp_mod_mp_mpp_er         196  mpp_util.inc
fms_lad2_xanadu.x  0000000001B96F83  fms_mod_mp_fms_er         589  fms.F90
fms_lad2_xanadu.x  0000000001F49F40  sat_vapor_pres_mo         609  sat_vapor_pres.F90
fms_lad2_xanadu.x  0000000000510C65  surface_flux_mod_         284  surface_flux.F90
fms_lad2_xanadu.x  000000000047FE8D  atm_land_ice_flux        1148  atm_land_ice_flux_exchange.F90
fms_lad2_xanadu.x  0000000002C5FF33  Unknown               Unknown  Unknown
fms_lad2_xanadu.x  0000000002C34B93  Unknown               Unknown  Unknown
fms_lad2_xanadu.x  0000000002C16FBF  Unknown               Unknown  Unknown
fms_lad2_xanadu.x  000000000046D5AC  atm_land_ice_flux        1133  atm_land_ice_flux_exchange.F90
fms_lad2_xanadu.x  000000000040C9B5  MAIN__                    816  coupler_main.F90

The problem is that the surface temperature goes outside the accepted range. The surface temperatures are also different between the new io and old io. I was able to trace the issue to amip_interp.F90. In amip_interp.F90, the data is read and then multiplied by 100 and converted to an integer.

```F90
     if ( interp_oi_sst ) then
          call read_data(fileobj, ncfieldname, tmp_dat, unlim_dim_level=k)
!     interpolate tmp_dat(360, 180) ---> dat(mobs,nobs) (to enable SST anom computation)
          if ( mobs/=360 .or. nobs/=180 ) then
               call a2a_bilinear(360, 180, tmp_dat, mobs, nobs, dat)
          else
               dat(:,:) = tmp_dat(:,:)
          endif
     else
          call read_data(fileobj, ncfieldname, dat, unlim_dim_level=k)
     endif
     idat =  nint(dat*100.)  ! reconstruct packed data for reproducibity
```

I compared 'dat' after it comes out after the read_data routine from the new io and old io and I saw that they were off by a factor of 100.

This is because in the old io after read_data is called:
It calls

1. fms/fms_io.F90:5996 read_data_3d_new
2. fms/fms_io.F90:5434 mpp_read
3. mpp/include/mpp_io_read.inc:98 read_record
4. mpp/include/mpp_read_2Ddecomp.h:233 READ_RECORD_CORE_

In READ_RECORD_CORE_, after it reads the data it multiplies it by field%scale (~0.01) and adds field%add:

```F90
             case(NF_SHORT)
                ptr1 = LOC(mpp_io_stack(1))
                error = NF_GET_VARA_INT2  ( mpp_file(unit)%ncid, field%id, start, axsiz, i2vals )
                call netcdf_err( error, mpp_file(unit), field=field )
                if(field%scale == 1.0 .and. field%add == 0.0) then
                   data(:)=i2vals(:)
                else
                   data(:)=i2vals(:)*field%scale + field%add
                end if
```

The new io does not check or use the scale factor or add_offset so the output after read_data is off by the scale factor. 

**To Reproduce**
Any model that uses the amip_interp routine with the new io will have this issue. 

**Expected behavior**
The new io should(?) check and use if a data file has the scale factor and add_offset. 

The amip_interp should not assume that the scale factor is always used and that it is always 100. 

As a quick fix, removing the *100 in [1471 on amip_interp.F90](https://github.com/NOAA-GFDL/FMS/blob/7e182ed7b6f36d02050c206b519df23c2bfc16ba/amip_interp/amip_interp.F90#L1471) will also solve the problem. 

**System Environment**
Describe the system environment, include:
All systems are affected

**Additional context**
The stdout can be found in GAEA: /lustre/f2/dev/gfdl/Uriel.Ramirez/land_xml/esm2mb-cold/stdout.ncrc3.intel16-debug-openmp/stdout/run/esm2mb-cold_1x3m0d_64x1a.o201623170

**Comment by uramirez8707** [comment]
@wrongkindofdoctor

**Comment by bensonr** [comment]
This is a change in behavior from old to new IO.  Are the scale and offset values part of the variable metadata?  If so, we need to understand their purpose for being in the files and expected use/application to the data.

**Comment by uramirez8707** [comment]
@bensonr The files do have the scale factor and add offset attributes, the new io just doesn't do anything with them when reading the variable (SST in this case).

**Comment by underwoo** [comment]
@wrongkindofdoctor is this still an issue, or was it fixed in #236?

**Comment by wrongkindofdoctor** [comment]
Fixed in #236
